### ISSUE TITLE
UTF-8 exports are not exported, but are still referenced

### ISSUE THREAD

**Issue by filipesilva** [ampproject/rollup-plugin-closure-compiler#240]
## What's the issue?

UTF-8 exports, such as `ɵɵfoo`, will not be exported from a module. Instead, they will be put on `window`. However, they are still imported, leading to errors such as `Uncaught SyntaxError: The requested module './router-03e289f6.js' does not provide an export named 'x'`.

## How do we reproduce the issue?

Given `rollup@1.27.8`, `@ampproject/rollup-plugin-closure-compiler@0.13.0`, and these files:
```
// rollup.config.js
import compiler from '@ampproject/rollup-plugin-closure-compiler';
const closureConfig = ;
export default {
  input: './repro/main',
  output: {
    dir: './dist/',
    format: 'esm',
  },
  treeshake: true,
  plugins: [
    compiler({ charset: 'UTF-8'})
  ]
};
```
```
// repro/main.js
import {ɵɵfoo, baz} from './common';
ɵɵfoo();
baz();
import('./lazy');
```
```
// repro/lazy.js
import { ɵɵbar } from './common';
ɵɵbar();
```
```
// repro/common.js
export function ɵɵfoo() {console.log('foo')};
export function ɵɵbar() {console.log('bar')};
export function baz() {console.log('baz')};
```

Running `rollup` will yield the following result:
```
// dist/main.js
import { ɵ as ɵɵfoo, b as baz } from './common-73678462.js';\u0275\u0275foo();baz();import("./lazy-e4f1fda2.js");
```
```
// dist/lazy-e4f1fda2.js
import { a as ɵɵbar } from './common-73678462.js';\u0275\u0275bar();
```
```
// dist/common-73678462.js
window["ɵɵbar"]=function(){console.log("bar")};var baz=function(){console.log("baz")};window["ɵɵfoo"]=function(){console.log("foo")};export{baz as b};
```

The `ɵ as ɵɵfoo` and `a as ɵɵbar` imports are present but these values are not exported from `common-73678462.js`.

## Additional context

Ideally the symbols would not go on window at all but just be exported instead.

**Comment by kristoferbaxter** [comment]
Let’s address this while adding support for code splitting post rollup 1.0.

**Comment by kristoferbaxter** [comment]
I attempted to reproduce this with the newly released `0.14.0` branch and was unable to.

Can you see if the latest version suffers from this issue? Also #250 adds tests for this case.

**Comment by filipesilva** [comment]
Heya, thanks for getting back to me!

I don't see the same problem as in the original repro anymore with version 0.18.1.

I do see a new error when trying to use the plugin in my real repo though.
```
git clone https://github.com/filipesilva/rollup-closure-cli
cd rollup-closure-cli
yarn 
yarn rollup-closure
```

This shows the following error:
```
kamik@RED-X1C6 MINGW64 /d/sandbox/rollup-closure-cli (master)
$ yarn rollup-closure
yarn run v1.17.3
$ node patch-bo-to-make-var-iife.js
$ rollup -c rollup-closure.config.js

./out-tsc/app/src/main-rollup.js → ./dist-rollup-closure/...
(node:15084) UnhandledPromiseRejectionWarning: SyntaxError: Identifier 'B' has already been declared (493:94)
    at _class.pp$4.raise (D:\sandbox\rollup-closure-cli\node_modules\acorn\dist\acorn.js:2836:15)
    at _class.pp$5.declareName (D:\sandbox\rollup-closure-cli\node_modules\acorn\dist\acorn.js:2910:28)
    at _class.pp$2.checkLVal (D:\sandbox\rollup-closure-cli\node_modules\acorn\dist\acorn.js:1816:77)
    at _class.pp$1.parseFunction (D:\sandbox\rollup-closure-cli\node_modules\acorn\dist\acorn.js:1270:16)
    at _class.pp$1.parseFunctionStatement (D:\sandbox\rollup-closure-cli\node_modules\acorn\dist\acorn.js:983:17)
    at _class.pp$1.parseStatement (D:\sandbox\rollup-closure-cli\node_modules\acorn\dist\acorn.js:830:19)
    at _class.parseStatement (D:\sandbox\rollup-closure-cli\node_modules\acorn-dynamic-import\lib\index.js:63:118)
    at _class.pp$1.parseTopLevel (D:\sandbox\rollup-closure-cli\node_modules\acorn\dist\acorn.js:746:23)
    at _class.parse (D:\sandbox\rollup-closure-cli\node_modules\acorn\dist\acorn.js:553:17)
    at Function.parse (D:\sandbox\rollup-closure-cli\node_modules\acorn\dist\acorn.js:576:37)
(node:15084) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejec
ting a promise which was not handled with .catch(). (rejection id: 2)
(node:15084) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process w
ith a non-zero exit code.
Done in 19.91s.
```

I can't really say with a lot of certainty it's related. But it started appearing after 0.14.0 (I tested with it too)  and it looks like it could be related since there is renaming going on for these vars. I don't know how to further debug and confirm it though.

**Comment by kristoferbaxter** [comment]
I’ll check this example out and enable debugging.

**Comment by kristoferbaxter** [comment]
This appears to be fixed with `0.19.0`. Please give it a go!

**Comment by filipesilva** [comment]
Can confirm it does, awesome! Thank you so much!
